yt/yt/client/table_client/composite_compare.cpp

```diff
@@ -148,6 +150,36 @@ Y_FORCE_INLINE static int CompareYsonItems(const TYsonItem& lhs, const TYsonItem
     return ComparePrimitive(static_cast<ui32>(lhsClass), static_cast<ui32>(rhsClass));
 }
 
+// Returns the minimum binary size that should be accounted for with this item.
+// Item should not be Map or Attribute-related, there is no special handling for them.
+i64 GetMinResultingSize(const TYsonItem& item, bool isInsideList)
+{
+    // These bytes were already accounted when handling the corresponding BeginList.
+    if (item.GetType() == EYsonItemType::EndList) {
+        return 0;
+    }
+
+    auto resultingSize = item.GetBinarySize();
+
+    // Strings can be truncated, so we only count the remaining bytes for now.
```

Comment:
Shouldn't we do something similar for nested lists?

Reply:
That's exactly what we are doing when we count BeginList as 2 or 3 bytes depending on whether there is an enclosing list.